src/diff/catch-error.js

```diff
@@ -7,26 +9,26 @@
  */
 export function _catchError(error, vnode) {
 	/** @type {import('../internal').Component} */
-	let component, ctor, handled;
+	let component, ctor;
+
+	if (vnode._component) vnode._component._mode = MODE_ERRORED;
```

Comment:
I don't think we did this at the other mode callsites yet, but this seems like a case where we need to account for the possibility of combined mode flags (maybe?):

```suggestion
	if (vnode._component) vnode._component._mode |= MODE_ERRORED;
```

If there's no combinations here yet we could punt on it for now - I know we'd agreed to only do bitfields via wrapper functions to keep the code clear.